Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -34,4 +49,13 @@ def env_apple
   def oracle_java_os
     :darwin
   end
+
+  def which_cask_needed?
+    if @version.nil? || @version.to_s.end_with?("+") ||
+       @version.to_f >= CURRENT_JAVA.to_f
+      JAVA_CASK_MAP.fetch(CURRENT_JAVA.to_s)
+    else
+      JAVA_CASK_MAP.fetch("1.8")
```

Comment:
A comment explaining the 9.0/1.8 situation just now would be great 👍

Reply:
👍